oort/oort-web/src/main/groovy/com/netflix/spinnaker/oort/controllers/ProjectController.groovy

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.oort.controllers
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.core.services.Front50Service
+import com.netflix.spinnaker.oort.model.Cluster
+import com.netflix.spinnaker.oort.model.ClusterProvider
+import com.netflix.spinnaker.oort.model.ServerGroup
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.MessageSource
+import org.springframework.context.i18n.LocaleContextHolder
+import org.springframework.http.HttpStatus
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
+import org.springframework.stereotype.Component
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import rx.Observable
+import rx.Scheduler
+import rx.schedulers.Schedulers
+
+import com.netflix.spinnaker.oort.model.ServerGroup.InstanceCounts as InstanceCounts
+
+@Slf4j
+@RestController
+@RequestMapping("/projects/{project}")
+class ProjectController {
+
+  private final Scheduler queryClusterScheduler
+
+  @Autowired
+  ProjectController(@Value('${threadPool.queryCluster:25}') int threadPoolSize) {
+    this(Schedulers.from(newFixedThreadPool(threadPoolSize)))
+  }
+
+  ProjectController(Scheduler queryClusterScheduler) {
+    this.queryClusterScheduler = queryClusterScheduler
+  }
+
+  @Autowired
+  Front50Service front50Service
+
+  @Autowired
+  MessageSource messageSource
+
+  @Autowired
+  List<ClusterProvider> clusterProviders
+
+  @RequestMapping(method= RequestMethod.GET, value = "/clusters")
+  List<ClusterModel> getClusters(@PathVariable String project) {
+    Map projectConfig = null
+    try {
+      projectConfig = front50Service.getProject(project)
+    } catch (e) {
+      log.error("Unable to fetch project (${project})", e)
+      throw new ProjectNotFoundException(name: project)
```

Comment:
Slightly different over there in that it's coming from a local provider vs. this one is going over the wire to front50. 

In theory we should just propagate the Retrofit error upwards, but we don't really have anything generic extracted to do that.

Guess this is fine for now.